### DISCUSSION TITLE
What algorithm is used in tri()?

### DISCUSSION THREAD

**Discussion by Jesse-Hamlin-Navias** [nesbox/TIC-80#2628] [Q&A]
I want to build a version of tri() that saves some info about what pixels the new triangle colored. So I basically want to copy and paste the tri() algorithm that is used, and sneak some extra lines in there, and use that instead of tri().

**Comment by joshgoebel** [comment]
Except `tri` is written in C, so unless you're compiling WASM/C you could only read it and rewrite it - not copy/paste.

**Reply by Jesse-Hamlin-Navias** [reply]
Lol yes I didn't expect to literally be able to copy paste. Just wondering if the algorithm used was a more common one with a name that I could look up and build from. Id rather stick with what is already in TIC-80 so that my triangles match perfectly with the systems triangles, if possible.

**Comment by cetebuk** [comment]
What's wrong with the source code? 
(v1.1.2837) file "draw.c" line 716 "tic_api_tri" and line 841 "tic_api_ttri".
Both use line 650 "drawTri" procedure.
